### ISSUE TITLE
Notice during exclusion pattern checking

### ISSUE THREAD

**Issue by aik099** [squizlabs/PHP_CodeSniffer#1214]
In my `ruleset.xml` I have this code:

```xml
<rule ref="Generic.WhiteSpace.ScopeIndent.IncorrectExact">
	<exclude-pattern>/core/</exclude-pattern>
	<exclude-pattern>\core\</exclude-pattern>
</rule>
```

The `/core/` part is for Linux users and the `\core\` part is for Windows users.

When the `\core\` is evaluated in https://github.com/squizlabs/PHP_CodeSniffer/blob/master/CodeSniffer/File.php#L1063 line the final regexp looks like: 

```
`\core\`i
```

where `\` acts as escaper to regular expression separator symbol and expression is treated as broken, which results in this notice for every processed sniff/error:

```
PHP Warning:  preg_match(): No ending delimiter '`' found in /home/user/.composer/vendor/squizlabs/php_codesniffer/CodeSniffer/File.php on line 1063
```

P.S.
Such approach worked in previous PHP_CodeSniffer versions.

**Comment by aik099** [comment]
My bad, just looked a few lines above in that file and noticed that if on Windows the `/` (linux path separators) are automatically converted to `\` (windows path separators).
